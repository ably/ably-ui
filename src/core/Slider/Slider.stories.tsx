import React, { ReactNode } from "react";
import Slider from "./component.tsx";

export default {
  title: "Components/Slider",
  component: Slider,
  parameters: {
    layout: "fullscreen",
  },
  tags: ["autodocs"],
};

const Slide = ({ children }: { children: ReactNode }) => (
  <div className="h-full p-24 bg-white rounded">
    <p className="ui-text-p2 text-center">{children}</p>
  </div>
);

const slides = [
  <Slide key="1">
    Powers live chat, updates, analytics, and composition for millions of users.
  </Slide>,
  <Slide key="2">
    Powers virtual venues for millions of event attendees around the world.
  </Slide>,
  <Slide key="3">
    Provides 5 million daily users with live financial commentary and stock
    tickers.
  </Slide>,
  <Slide key="4">Monitors live car performance data across the USA.</Slide>,
];

export const SliderOnAllBreakpoints = {
  args: {
    slides,
  },
};

export const SliderOnSmallBreakpointOnly = {
  args: {
    slides,
    classes: `sm:grid-cols-${slides.length / 2} md:grid-cols-${slides.length}`,
    mqEnableThreshold: () => !window.matchMedia("(min-width: 48rem)").matches,
  },
};
