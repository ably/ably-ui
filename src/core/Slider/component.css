.ui-slider-marker {
  font-size: 0.5rem;
  top: -1px;

  @apply leading-none px-4 relative;
}

@keyframes fillAnimation {
  0% {
    width: 0%;
  }
  100% {
    width: 100%;
  }
}

.ui-icon-cta {
  @apply cursor-pointer overflow-hidden;
  @apply rounded border-2 border-mid-grey hover:border-active-orange;
  transition: all 0.4s;
}

@screen sm {
  .ui-icon-cta-left:hover .ui-icon-cta-holder {
    transform: translateX(-120%);
  }
  .ui-icon-cta-right .ui-icon-cta-holder {
    transform: translateX(-120%);
  }
  .ui-icon-cta-right:hover .ui-icon-cta-holder {
    transform: translateX(0%);
  }
}

.ui-icon-cta-holder {
  @apply w-full h-full;
  transition: all 0.4s;
}
