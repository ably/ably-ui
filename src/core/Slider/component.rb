module AblyUi
  module Core
    class Slider < ViewComponent::Base
    end
  end
end
