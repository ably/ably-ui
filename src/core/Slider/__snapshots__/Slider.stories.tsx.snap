// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JS Components/Slider FloatingControlPosition smoke-test 1`] = `
<div class="relative">
  <div class="overflow-y-visible overflow-x-clip w-full py-40">
    <div class="flex items-center  "
         style="transform: translateX(-100%);"
    >
      <div class="w-full flex-shrink-0 flex justify-center sm:px-60">
        <div class="relative ">
          <div class="relative w-full sm:w-[560px] md:w-[784px] lg:w-[960px] bg-white overflow-hidden  flex gap-40 rounded-3xl shadow-container-subtle">
            <div class="w-full md:w-2/3 flex flex-col gap-24 pr-40 md:pr-0 pt-40 pl-40 pb-40 sm:pb-[120px] md:pb-40">
              <h2 class="ui-text-h2 font-medium text-neutral-1000">
                “Ably seamlessly absorbs sudden bursts in load during unexpected client events. The integration was easy and we were live in under a month.”
              </h2>
              <div class="flex flex-col sm:flex-row gap-32">
                <div class="flex gap-8">
                  <div class="static self-center sm:absolute sm:-bottom-48 sm:-right-[56px] rounded-full bg-gradient-to-l from-neutral-200 to-50% to-neutral-500 w-[48px] h-[48px] sm:w-[201px] sm:h-[201px] md:w-[257px] md:h-[257px] lg:w-[280px] lg:h-[280px] overflow-hidden flex items-center justify-center sm:border-[16px] border-neutral-200">
                    <img src="https://picsum.photos/id/64/200"
                         alt="test-image"
                    >
                  </div>
                  <div class="sm:py-16">
                    <p class="ui-text-p1 text-neutral-1300">
                      Aleksandar Kostadinov
                    </p>
                    <p class="ui-text-p3 text-neutral-800">
                      Co-Founder &amp; Technical Leader
                    </p>
                  </div>
                </div>
                <div class="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500">
                </div>
                <div class="flex items-center gap-4">
                  <img src="https://picsum.photos/id/145/40"
                       alt="test-image"
                  >
                  <p class="ui-text-h4 font-bold">
                    Mentimeter
                  </p>
                </div>
              </div>
              <a href="/case-study"
                 class="ui-btn self-start"
              >
                Read case study
                <svg class="ml-4"
                     style="width: 1.25rem; height: 1.25rem;"
                >
                  <use xlink:href="#sprite-icon-gui-arrow-right">
                  </use>
                </svg>
              </a>
            </div>
          </div>
          <div class="absolute h-256 -z-10 -bottom-48 -left-36 w-1/5 rounded-full blur-xl opacity-50 transform -rotate-45 bg-gradient-to-bl from-bg-glow-green to-bg-glow-teal">
          </div>
          <div class="absolute h-256 -z-10 -top-48 -right-48 w-3/5 rounded-full blur-xl opacity-50 transform rotate-12 bg-gradient-to-br from-bg-glow-green to-bg-glow-teal">
          </div>
        </div>
      </div>
      <div class="w-full flex-shrink-0 flex justify-center sm:px-60">
        <div class="relative ">
          <div class="relative w-full sm:w-[560px] md:w-[784px] lg:w-[960px] bg-white overflow-hidden  flex gap-40 rounded-3xl shadow-container-subtle">
            <div class="w-full md:w-2/3 flex flex-col gap-24 pr-40 md:pr-0 pt-40 pl-40 pb-40 sm:pb-[120px] md:pb-40">
              <h2 class="ui-text-h2 font-medium text-neutral-1000">
                “Ably seamlessly absorbs sudden bursts in load during unexpected client events. The integration was easy and we were live in under a month.”
              </h2>
              <div class="flex flex-col sm:flex-row gap-32">
                <div class="flex gap-8">
                  <div class="static self-center sm:absolute sm:-bottom-48 sm:-right-[56px] rounded-full bg-gradient-to-l from-neutral-200 to-50% to-neutral-500 w-[48px] h-[48px] sm:w-[201px] sm:h-[201px] md:w-[257px] md:h-[257px] lg:w-[280px] lg:h-[280px] overflow-hidden flex items-center justify-center sm:border-[16px] border-neutral-200">
                    <img src="https://picsum.photos/id/64/200"
                         alt="test-image"
                    >
                  </div>
                  <div class="sm:py-16">
                    <p class="ui-text-p1 text-neutral-1300">
                      Johan Bengtsson
                    </p>
                    <p class="ui-text-p3 text-neutral-800">
                      Co-Founder &amp; Technical Leader
                    </p>
                  </div>
                </div>
                <div class="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500">
                </div>
                <div class="flex items-center gap-4">
                  <img src="https://picsum.photos/id/145/40"
                       alt="test-image"
                  >
                  <p class="ui-text-h4 font-bold">
                    Mentimeter
                  </p>
                </div>
              </div>
              <a href="/case-study"
                 class="ui-btn self-start"
              >
                Read case study
                <svg class="ml-4"
                     style="width: 1.25rem; height: 1.25rem;"
                >
                  <use xlink:href="#sprite-icon-gui-arrow-right">
                  </use>
                </svg>
              </a>
            </div>
          </div>
          <div class="absolute h-256 -z-10 -bottom-48 -left-36 w-1/5 rounded-full blur-xl opacity-50 transform -rotate-45 bg-gradient-to-bl from-bg-glow-green to-bg-glow-teal">
          </div>
          <div class="absolute h-256 -z-10 -top-48 -right-48 w-3/5 rounded-full blur-xl opacity-50 transform rotate-12 bg-gradient-to-br from-bg-glow-green to-bg-glow-teal">
          </div>
        </div>
      </div>
      <div class="w-full flex-shrink-0 flex justify-center sm:px-60">
        <div class="relative ">
          <div class="relative w-full sm:w-[560px] md:w-[784px] lg:w-[960px] bg-white overflow-hidden  flex gap-40 rounded-3xl shadow-container-subtle">
            <div class="w-full md:w-2/3 flex flex-col gap-24 pr-40 md:pr-0 pt-40 pl-40 pb-40 sm:pb-[120px] md:pb-40">
              <h2 class="ui-text-h2 font-medium text-neutral-1000">
                “Ably seamlessly absorbs sudden bursts in load during unexpected client events. The integration was easy and we were live in under a month.”
              </h2>
              <div class="flex flex-col sm:flex-row gap-32">
                <div class="flex gap-8">
                  <div class="static self-center sm:absolute sm:-bottom-48 sm:-right-[56px] rounded-full bg-gradient-to-l from-neutral-200 to-50% to-neutral-500 w-[48px] h-[48px] sm:w-[201px] sm:h-[201px] md:w-[257px] md:h-[257px] lg:w-[280px] lg:h-[280px] overflow-hidden flex items-center justify-center sm:border-[16px] border-neutral-200">
                    <img src="https://picsum.photos/id/64/200"
                         alt="test-image"
                    >
                  </div>
                  <div class="sm:py-16">
                    <p class="ui-text-p1 text-neutral-1300">
                      Mirko Bergman
                    </p>
                    <p class="ui-text-p3 text-neutral-800">
                      Co-Founder &amp; Technical Leader
                    </p>
                  </div>
                </div>
                <div class="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500">
                </div>
                <div class="flex items-center gap-4">
                  <img src="https://picsum.photos/id/145/40"
                       alt="test-image"
                  >
                  <p class="ui-text-h4 font-bold">
                    Mentimeter
                  </p>
                </div>
              </div>
              <a href="/case-study"
                 class="ui-btn self-start"
              >
                Read case study
                <svg class="ml-4"
                     style="width: 1.25rem; height: 1.25rem;"
                >
                  <use xlink:href="#sprite-icon-gui-arrow-right">
                  </use>
                </svg>
              </a>
            </div>
          </div>
          <div class="absolute h-256 -z-10 -bottom-48 -left-36 w-1/5 rounded-full blur-xl opacity-50 transform -rotate-45 bg-gradient-to-bl from-bg-glow-green to-bg-glow-teal">
          </div>
          <div class="absolute h-256 -z-10 -top-48 -right-48 w-3/5 rounded-full blur-xl opacity-50 transform rotate-12 bg-gradient-to-br from-bg-glow-green to-bg-glow-teal">
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="flex items-center pointer-events-none sm:flex sm:absolute inset-0 justify-between">
    <button class="hidden sm:flex w-48 h-48 pointer-events-auto rounded border border-mid-grey hover:border-active-orange flex justify-center items-center ui-icon-cta ui-icon-cta-left">
      <div class="ui-icon-cta-holder flex w-48">
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-left">
            </use>
          </svg>
        </div>
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-left">
            </use>
          </svg>
        </div>
      </div>
    </button>
    <ul class="flex gap-4 left-1/2 absolute bottom-0 transform -translate-x-1/2">
      <li class="relative w-40 h-4 mx-1 rounded-full bg-neutral-500">
        <span class="absolute inset-0 rounded-full bg-active-orange"
              style="animation: 10000ms linear 0s 1 normal none running fillAnimation;"
        >
        </span>
      </li>
      <li class="relative w-40 h-4 mx-1 rounded-full bg-neutral-500">
      </li>
      <li class="relative w-40 h-4 mx-1 rounded-full bg-neutral-500">
      </li>
      <li class="relative w-40 h-4 mx-1 rounded-full bg-neutral-500">
      </li>
    </ul>
    <button class="hidden sm:flex w-48 h-48 pointer-events-auto rounded border border-mid-grey hover:border-active-orange justify-center items-center ui-icon-cta ui-icon-cta-right">
      <div class="ui-icon-cta-holder flex w-48">
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-right">
            </use>
          </svg>
        </div>
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-right">
            </use>
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`JS Components/Slider InlineControlPosition smoke-test 1`] = `
<div class="relative">
  <div class="overflow-y-visible overflow-x-clip w-full py-40">
    <div class="flex items-center  "
         style="transform: translateX(-100%);"
    >
      <div class="w-full flex-shrink-0 flex justify-center sm:px-60">
        <div class="relative ">
          <div class="relative w-full sm:w-[560px] md:w-[784px] lg:w-[960px] bg-white overflow-hidden  flex gap-40 rounded-3xl shadow-container-subtle">
            <div class="w-full md:w-2/3 flex flex-col gap-24 pr-40 md:pr-0 pt-40 pl-40 pb-40 sm:pb-[120px] md:pb-40">
              <h2 class="ui-text-h2 font-medium text-neutral-1000">
                “Ably seamlessly absorbs sudden bursts in load during unexpected client events. The integration was easy and we were live in under a month.”
              </h2>
              <div class="flex flex-col sm:flex-row gap-32">
                <div class="flex gap-8">
                  <div class="static self-center sm:absolute sm:-bottom-48 sm:-right-[56px] rounded-full bg-gradient-to-l from-neutral-200 to-50% to-neutral-500 w-[48px] h-[48px] sm:w-[201px] sm:h-[201px] md:w-[257px] md:h-[257px] lg:w-[280px] lg:h-[280px] overflow-hidden flex items-center justify-center sm:border-[16px] border-neutral-200">
                    <img src="https://picsum.photos/id/64/200"
                         alt="test-image"
                    >
                  </div>
                  <div class="sm:py-16">
                    <p class="ui-text-p1 text-neutral-1300">
                      Aleksandar Kostadinov
                    </p>
                    <p class="ui-text-p3 text-neutral-800">
                      Co-Founder &amp; Technical Leader
                    </p>
                  </div>
                </div>
                <div class="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500">
                </div>
                <div class="flex items-center gap-4">
                  <img src="https://picsum.photos/id/145/40"
                       alt="test-image"
                  >
                  <p class="ui-text-h4 font-bold">
                    Mentimeter
                  </p>
                </div>
              </div>
              <a href="/case-study"
                 class="ui-btn self-start"
              >
                Read case study
                <svg class="ml-4"
                     style="width: 1.25rem; height: 1.25rem;"
                >
                  <use xlink:href="#sprite-icon-gui-arrow-right">
                  </use>
                </svg>
              </a>
            </div>
          </div>
          <div class="absolute h-256 -z-10 -bottom-48 -left-36 w-1/5 rounded-full blur-xl opacity-50 transform -rotate-45 bg-gradient-to-bl from-bg-glow-green to-bg-glow-teal">
          </div>
          <div class="absolute h-256 -z-10 -top-48 -right-48 w-3/5 rounded-full blur-xl opacity-50 transform rotate-12 bg-gradient-to-br from-bg-glow-green to-bg-glow-teal">
          </div>
        </div>
      </div>
      <div class="w-full flex-shrink-0 flex justify-center sm:px-60">
        <div class="relative ">
          <div class="relative w-full sm:w-[560px] md:w-[784px] lg:w-[960px] bg-white overflow-hidden  flex gap-40 rounded-3xl shadow-container-subtle">
            <div class="w-full md:w-2/3 flex flex-col gap-24 pr-40 md:pr-0 pt-40 pl-40 pb-40 sm:pb-[120px] md:pb-40">
              <h2 class="ui-text-h2 font-medium text-neutral-1000">
                “Ably seamlessly absorbs sudden bursts in load during unexpected client events. The integration was easy and we were live in under a month.”
              </h2>
              <div class="flex flex-col sm:flex-row gap-32">
                <div class="flex gap-8">
                  <div class="static self-center sm:absolute sm:-bottom-48 sm:-right-[56px] rounded-full bg-gradient-to-l from-neutral-200 to-50% to-neutral-500 w-[48px] h-[48px] sm:w-[201px] sm:h-[201px] md:w-[257px] md:h-[257px] lg:w-[280px] lg:h-[280px] overflow-hidden flex items-center justify-center sm:border-[16px] border-neutral-200">
                    <img src="https://picsum.photos/id/64/200"
                         alt="test-image"
                    >
                  </div>
                  <div class="sm:py-16">
                    <p class="ui-text-p1 text-neutral-1300">
                      Johan Bengtsson
                    </p>
                    <p class="ui-text-p3 text-neutral-800">
                      Co-Founder &amp; Technical Leader
                    </p>
                  </div>
                </div>
                <div class="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500">
                </div>
                <div class="flex items-center gap-4">
                  <img src="https://picsum.photos/id/145/40"
                       alt="test-image"
                  >
                  <p class="ui-text-h4 font-bold">
                    Mentimeter
                  </p>
                </div>
              </div>
              <a href="/case-study"
                 class="ui-btn self-start"
              >
                Read case study
                <svg class="ml-4"
                     style="width: 1.25rem; height: 1.25rem;"
                >
                  <use xlink:href="#sprite-icon-gui-arrow-right">
                  </use>
                </svg>
              </a>
            </div>
          </div>
          <div class="absolute h-256 -z-10 -bottom-48 -left-36 w-1/5 rounded-full blur-xl opacity-50 transform -rotate-45 bg-gradient-to-bl from-bg-glow-green to-bg-glow-teal">
          </div>
          <div class="absolute h-256 -z-10 -top-48 -right-48 w-3/5 rounded-full blur-xl opacity-50 transform rotate-12 bg-gradient-to-br from-bg-glow-green to-bg-glow-teal">
          </div>
        </div>
      </div>
      <div class="w-full flex-shrink-0 flex justify-center sm:px-60">
        <div class="relative ">
          <div class="relative w-full sm:w-[560px] md:w-[784px] lg:w-[960px] bg-white overflow-hidden  flex gap-40 rounded-3xl shadow-container-subtle">
            <div class="w-full md:w-2/3 flex flex-col gap-24 pr-40 md:pr-0 pt-40 pl-40 pb-40 sm:pb-[120px] md:pb-40">
              <h2 class="ui-text-h2 font-medium text-neutral-1000">
                “Ably seamlessly absorbs sudden bursts in load during unexpected client events. The integration was easy and we were live in under a month.”
              </h2>
              <div class="flex flex-col sm:flex-row gap-32">
                <div class="flex gap-8">
                  <div class="static self-center sm:absolute sm:-bottom-48 sm:-right-[56px] rounded-full bg-gradient-to-l from-neutral-200 to-50% to-neutral-500 w-[48px] h-[48px] sm:w-[201px] sm:h-[201px] md:w-[257px] md:h-[257px] lg:w-[280px] lg:h-[280px] overflow-hidden flex items-center justify-center sm:border-[16px] border-neutral-200">
                    <img src="https://picsum.photos/id/64/200"
                         alt="test-image"
                    >
                  </div>
                  <div class="sm:py-16">
                    <p class="ui-text-p1 text-neutral-1300">
                      Mirko Bergman
                    </p>
                    <p class="ui-text-p3 text-neutral-800">
                      Co-Founder &amp; Technical Leader
                    </p>
                  </div>
                </div>
                <div class="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500">
                </div>
                <div class="flex items-center gap-4">
                  <img src="https://picsum.photos/id/145/40"
                       alt="test-image"
                  >
                  <p class="ui-text-h4 font-bold">
                    Mentimeter
                  </p>
                </div>
              </div>
              <a href="/case-study"
                 class="ui-btn self-start"
              >
                Read case study
                <svg class="ml-4"
                     style="width: 1.25rem; height: 1.25rem;"
                >
                  <use xlink:href="#sprite-icon-gui-arrow-right">
                  </use>
                </svg>
              </a>
            </div>
          </div>
          <div class="absolute h-256 -z-10 -bottom-48 -left-36 w-1/5 rounded-full blur-xl opacity-50 transform -rotate-45 bg-gradient-to-bl from-bg-glow-green to-bg-glow-teal">
          </div>
          <div class="absolute h-256 -z-10 -top-48 -right-48 w-3/5 rounded-full blur-xl opacity-50 transform rotate-12 bg-gradient-to-br from-bg-glow-green to-bg-glow-teal">
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="flex items-center pointer-events-none ui-standard-container justify-center gap-24 -mt-16">
    <button class="w-32 h-32 pointer-events-auto rounded border border-mid-grey hover:border-active-orange flex justify-center items-center ui-icon-cta ui-icon-cta-left">
      <div class="ui-icon-cta-holder flex w-48">
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-left">
            </use>
          </svg>
        </div>
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-left">
            </use>
          </svg>
        </div>
      </div>
    </button>
    <ul class="flex gap-4 left-1/2 bottom-0">
      <li class="relative w-40 h-4 mx-1 rounded-full bg-neutral-500">
        <span class="absolute inset-0 rounded-full bg-active-orange"
              style="animation: 10000ms linear 0s 1 normal none running fillAnimation;"
        >
        </span>
      </li>
      <li class="relative w-40 h-4 mx-1 rounded-full bg-neutral-500">
      </li>
      <li class="relative w-40 h-4 mx-1 rounded-full bg-neutral-500">
      </li>
      <li class="relative w-40 h-4 mx-1 rounded-full bg-neutral-500">
      </li>
    </ul>
    <button class="w-32 h-32 pointer-events-auto rounded border border-mid-grey hover:border-active-orange justify-center items-center ui-icon-cta ui-icon-cta-right">
      <div class="ui-icon-cta-holder flex w-48">
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-right">
            </use>
          </svg>
        </div>
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-right">
            </use>
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`JS Components/Slider WithoutIntervalIndicator smoke-test 1`] = `
<div class="relative">
  <div class="overflow-y-visible overflow-x-clip w-full py-40">
    <div class="flex items-center  "
         style="transform: translateX(-100%);"
    >
      <div class="w-full flex-shrink-0 flex justify-center sm:px-60">
        <div class="relative ">
          <div class="relative w-full sm:w-[560px] md:w-[784px] lg:w-[960px] bg-white overflow-hidden  flex gap-40 rounded-3xl shadow-container-subtle">
            <div class="w-full md:w-2/3 flex flex-col gap-24 pr-40 md:pr-0 pt-40 pl-40 pb-40 sm:pb-[120px] md:pb-40">
              <h2 class="ui-text-h2 font-medium text-neutral-1000">
                “Ably seamlessly absorbs sudden bursts in load during unexpected client events. The integration was easy and we were live in under a month.”
              </h2>
              <div class="flex flex-col sm:flex-row gap-32">
                <div class="flex gap-8">
                  <div class="static self-center sm:absolute sm:-bottom-48 sm:-right-[56px] rounded-full bg-gradient-to-l from-neutral-200 to-50% to-neutral-500 w-[48px] h-[48px] sm:w-[201px] sm:h-[201px] md:w-[257px] md:h-[257px] lg:w-[280px] lg:h-[280px] overflow-hidden flex items-center justify-center sm:border-[16px] border-neutral-200">
                    <img src="https://picsum.photos/id/64/200"
                         alt="test-image"
                    >
                  </div>
                  <div class="sm:py-16">
                    <p class="ui-text-p1 text-neutral-1300">
                      Aleksandar Kostadinov
                    </p>
                    <p class="ui-text-p3 text-neutral-800">
                      Co-Founder &amp; Technical Leader
                    </p>
                  </div>
                </div>
                <div class="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500">
                </div>
                <div class="flex items-center gap-4">
                  <img src="https://picsum.photos/id/145/40"
                       alt="test-image"
                  >
                  <p class="ui-text-h4 font-bold">
                    Mentimeter
                  </p>
                </div>
              </div>
              <a href="/case-study"
                 class="ui-btn self-start"
              >
                Read case study
                <svg class="ml-4"
                     style="width: 1.25rem; height: 1.25rem;"
                >
                  <use xlink:href="#sprite-icon-gui-arrow-right">
                  </use>
                </svg>
              </a>
            </div>
          </div>
          <div class="absolute h-256 -z-10 -bottom-48 -left-36 w-1/5 rounded-full blur-xl opacity-50 transform -rotate-45 bg-gradient-to-bl from-bg-glow-green to-bg-glow-teal">
          </div>
          <div class="absolute h-256 -z-10 -top-48 -right-48 w-3/5 rounded-full blur-xl opacity-50 transform rotate-12 bg-gradient-to-br from-bg-glow-green to-bg-glow-teal">
          </div>
        </div>
      </div>
      <div class="w-full flex-shrink-0 flex justify-center sm:px-60">
        <div class="relative ">
          <div class="relative w-full sm:w-[560px] md:w-[784px] lg:w-[960px] bg-white overflow-hidden  flex gap-40 rounded-3xl shadow-container-subtle">
            <div class="w-full md:w-2/3 flex flex-col gap-24 pr-40 md:pr-0 pt-40 pl-40 pb-40 sm:pb-[120px] md:pb-40">
              <h2 class="ui-text-h2 font-medium text-neutral-1000">
                “Ably seamlessly absorbs sudden bursts in load during unexpected client events. The integration was easy and we were live in under a month.”
              </h2>
              <div class="flex flex-col sm:flex-row gap-32">
                <div class="flex gap-8">
                  <div class="static self-center sm:absolute sm:-bottom-48 sm:-right-[56px] rounded-full bg-gradient-to-l from-neutral-200 to-50% to-neutral-500 w-[48px] h-[48px] sm:w-[201px] sm:h-[201px] md:w-[257px] md:h-[257px] lg:w-[280px] lg:h-[280px] overflow-hidden flex items-center justify-center sm:border-[16px] border-neutral-200">
                    <img src="https://picsum.photos/id/64/200"
                         alt="test-image"
                    >
                  </div>
                  <div class="sm:py-16">
                    <p class="ui-text-p1 text-neutral-1300">
                      Johan Bengtsson
                    </p>
                    <p class="ui-text-p3 text-neutral-800">
                      Co-Founder &amp; Technical Leader
                    </p>
                  </div>
                </div>
                <div class="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500">
                </div>
                <div class="flex items-center gap-4">
                  <img src="https://picsum.photos/id/145/40"
                       alt="test-image"
                  >
                  <p class="ui-text-h4 font-bold">
                    Mentimeter
                  </p>
                </div>
              </div>
              <a href="/case-study"
                 class="ui-btn self-start"
              >
                Read case study
                <svg class="ml-4"
                     style="width: 1.25rem; height: 1.25rem;"
                >
                  <use xlink:href="#sprite-icon-gui-arrow-right">
                  </use>
                </svg>
              </a>
            </div>
          </div>
          <div class="absolute h-256 -z-10 -bottom-48 -left-36 w-1/5 rounded-full blur-xl opacity-50 transform -rotate-45 bg-gradient-to-bl from-bg-glow-green to-bg-glow-teal">
          </div>
          <div class="absolute h-256 -z-10 -top-48 -right-48 w-3/5 rounded-full blur-xl opacity-50 transform rotate-12 bg-gradient-to-br from-bg-glow-green to-bg-glow-teal">
          </div>
        </div>
      </div>
      <div class="w-full flex-shrink-0 flex justify-center sm:px-60">
        <div class="relative ">
          <div class="relative w-full sm:w-[560px] md:w-[784px] lg:w-[960px] bg-white overflow-hidden  flex gap-40 rounded-3xl shadow-container-subtle">
            <div class="w-full md:w-2/3 flex flex-col gap-24 pr-40 md:pr-0 pt-40 pl-40 pb-40 sm:pb-[120px] md:pb-40">
              <h2 class="ui-text-h2 font-medium text-neutral-1000">
                “Ably seamlessly absorbs sudden bursts in load during unexpected client events. The integration was easy and we were live in under a month.”
              </h2>
              <div class="flex flex-col sm:flex-row gap-32">
                <div class="flex gap-8">
                  <div class="static self-center sm:absolute sm:-bottom-48 sm:-right-[56px] rounded-full bg-gradient-to-l from-neutral-200 to-50% to-neutral-500 w-[48px] h-[48px] sm:w-[201px] sm:h-[201px] md:w-[257px] md:h-[257px] lg:w-[280px] lg:h-[280px] overflow-hidden flex items-center justify-center sm:border-[16px] border-neutral-200">
                    <img src="https://picsum.photos/id/64/200"
                         alt="test-image"
                    >
                  </div>
                  <div class="sm:py-16">
                    <p class="ui-text-p1 text-neutral-1300">
                      Mirko Bergman
                    </p>
                    <p class="ui-text-p3 text-neutral-800">
                      Co-Founder &amp; Technical Leader
                    </p>
                  </div>
                </div>
                <div class="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500">
                </div>
                <div class="flex items-center gap-4">
                  <img src="https://picsum.photos/id/145/40"
                       alt="test-image"
                  >
                  <p class="ui-text-h4 font-bold">
                    Mentimeter
                  </p>
                </div>
              </div>
              <a href="/case-study"
                 class="ui-btn self-start"
              >
                Read case study
                <svg class="ml-4"
                     style="width: 1.25rem; height: 1.25rem;"
                >
                  <use xlink:href="#sprite-icon-gui-arrow-right">
                  </use>
                </svg>
              </a>
            </div>
          </div>
          <div class="absolute h-256 -z-10 -bottom-48 -left-36 w-1/5 rounded-full blur-xl opacity-50 transform -rotate-45 bg-gradient-to-bl from-bg-glow-green to-bg-glow-teal">
          </div>
          <div class="absolute h-256 -z-10 -top-48 -right-48 w-3/5 rounded-full blur-xl opacity-50 transform rotate-12 bg-gradient-to-br from-bg-glow-green to-bg-glow-teal">
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="flex items-center pointer-events-none sm:flex sm:absolute inset-0 justify-between">
    <button class="hidden sm:flex w-48 h-48 pointer-events-auto rounded border border-mid-grey hover:border-active-orange flex justify-center items-center ui-icon-cta ui-icon-cta-left">
      <div class="ui-icon-cta-holder flex w-48">
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-left">
            </use>
          </svg>
        </div>
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-left">
            </use>
          </svg>
        </div>
      </div>
    </button>
    <ul class="flex gap-4 left-1/2 absolute bottom-0 transform -translate-x-1/2">
      <li>
        <span class="ui-slider-marker text-active-orange"
              data-id="slider-marker"
        >
          ⬤
        </span>
      </li>
      <li>
        <span class="ui-slider-marker text-cool-black"
              data-id="slider-marker"
        >
          ⬤
        </span>
      </li>
      <li>
        <span class="ui-slider-marker text-cool-black"
              data-id="slider-marker"
        >
          ⬤
        </span>
      </li>
      <li>
        <span class="ui-slider-marker text-cool-black"
              data-id="slider-marker"
        >
          ⬤
        </span>
      </li>
    </ul>
    <button class="hidden sm:flex w-48 h-48 pointer-events-auto rounded border border-mid-grey hover:border-active-orange justify-center items-center ui-icon-cta ui-icon-cta-right">
      <div class="ui-icon-cta-holder flex w-48">
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-right">
            </use>
          </svg>
        </div>
        <div class="w-full h-full flex-shrink-0 flex items-center justify-center">
          <svg class
               style="width: 1.5rem; height: 1.5rem;"
          >
            <use xlink:href="#sprite-icon-gui-arrow-right">
            </use>
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;
